### PR TITLE
Drop EOL Python support and update dependencies

### PR DESCRIPTION
**Pull request by tucked**
Full test run to come...

### CODE REVIEW

**Comment by lfrancke** [comment]
This is a bit selfish but: Do you think this will be merged/done until next week? We've got an installation next week and it'd be great to know if I need to prepare a custom build or not.

**Comment by tucked** [comment]
@lfrancke, getting a full test run can take some time (because it involves running against all the supported OneFS versions), so it's kind of hard to say... If it's not ready by the end of the week, though, we can make a pre-release for you! Do you need #105 and #106 too?

File: src/isilon_hadoop_tools/onefs.py [hunk 1]
```diff
@@ -577,7 +577,9 @@ def _decorated(self, *args, **kwargs):
                     # https://github.com/Isilon/isilon_sdk_python/issues/14
                     raise_from(OneFSCertificateError, exc)
                 raise_from(OneFSConnectionError, exc)
-            except self._sdk.rest.ApiException as exc:  # pylint: disable=protected-access
+            except (
+                self._sdk.rest.ApiException
+            ) as exc:  # pylint: disable=protected-access
```
Comment (tucked) [inline src/isilon_hadoop_tools/onefs.py]: Wrong line?

**Comment by lfrancke** [comment]
Thank you but don't worry, no extra effort needed for us.
The other two we already did manually, this was mostly for the benefit of others after us
 I can try to get them updated next week.

**Comment by tucked** [comment]
Passing run on https://github.com/Isilon/isilon_hadoop_tools/pull/109/commits/4d0a84402137c4089f90f9523230624ffcf568cc: https://gist.github.com/tucked/2c1a6b845db63fc39430b0b5ac4e784d

**Review by teajason — APPROVED** [review]
Looks good, send it!

**Pull request merged**